Sources/MapboxDirections/OfflineDirections.swift

```diff
@@ -54,14 +56,25 @@ extension Directions: OfflineDirectionsProtocol {
     }
 
     /**
-     Fetch the available versions in descending chronological order. A version is represented as a String (yyyy-MM-dd or yyyy-MM-dd-x).
+     Fetches the available offline routing tile versions and returns them in descending chronological order. The most recent version should typically be passed into `downloadTiles(in:version:completionHandler:)`.
      
-     - parameter completionHandler: The closure to call with the results
-     - returns: A `URLSessionDataTask`
+     - parameter completionHandler: A closure of type `OfflineVersionsHandler` which will be called when the request completes
      */
     @discardableResult
-    @objc public func fetchAvailableOfflineVersions(completionHandler: @escaping OfflineVersionsHandler) -> URLSessionDataTask {
-        
+    public func fetchAvailableOfflineVersions(completionHandler: @escaping OfflineVersionsHandler) -> URLSessionDataTask {
+        let task = availableVersionsDataTask(completionHandler: completionHandler)
+        task.resume()
+        return task
+    }
+
+    /**
+     Creates and returns a URLSessionDataTask for downloading available offline routing tile versions. The caller is responsible for calling `-resume` on the task.
+     
+     - parameter completionHandler: A closure of type `OfflineVersionsHandler` which will be called when the request completes
+     */
+    @discardableResult
+    @objc(availableVersionsDataTaskWithCompletionHandler:)
+    public func availableVersionsDataTask(completionHandler: @escaping OfflineVersionsHandler) -> URLSessionDataTask {
         let task = URLSession.shared.dataTask(with: availableVersionsURL) { (data, response, error) in
             if let error = error {
                 return completionHandler(nil, error)
```

Comment:
`availableVersionsDataTask(_:)` still confuses me; I don’t really see the point in keeping it separate. In general, for any kind of API request, the developer would only ever need two kinds of methods:

* A method that encapsulates requesting and processing a response, taking a completion handler and returning a (discardable) task for cancellation.
* A method or property that provides access to the request payload itself without side effects.

These requirements are satisfied by `fetchAvailableOfflineVersions(_:)` and `availableVersionsURL`, and by `downloadTiles(in:version:)` and `tilesURL(for:version:)`. `availableVersionsDataTask(_:)` and `routingTilesDataTaskWith(_:version:completionHandler:)` are confusing hybrids, because they look like accessors or factory methods but take completion handlers, as if they’re the methods that perform the task. In fact, `completionHandler` is documented as being “called when the request completes”, but calling the method by itself doesn’t cause the completion handler to ever get called.

We can leave these methods in, but we should probably remove `@discardableResult` because the methods do nothing on their own.

Reply:
I agree. The hybrid methods don't improve readability or versatility so I folded them into their counterparts.